Std/Data/BitVec/Bitblast.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2023 by the authors listed in the file AUTHORS and their
+institutional affiliations. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Harun Khan, Abdalrhman M Mohamed
+-/
+import Std.Data.BitVec.Basic
+import Std.Data.Nat.Lemmas
+import Std.Data.Bool
+
+namespace Std
+open Nat Bool
+
+/-! ### Preliminaries -/
+
+theorem bodd_eq_bodd_iff {m n} : bodd n = bodd m ↔ n % 2 = m % 2 := by
+  cases hn : bodd n <;> cases hm : bodd m
+  <;> simp [mod_two_of_bodd, hn, hm]
+
+theorem testBit_two_pow_mul_add {n b w i} (h : i < w) :
+    Nat.testBit (2 ^ w * b + n) i = Nat.testBit n i := by
+  simp only [testBit, shiftRight_eq_div_pow, bodd_eq_bodd_iff]
+  rw [← Nat.add_sub_of_le (succ_le_of_lt h), Nat.succ_eq_add_one, Nat.add_assoc]
+  rw [Nat.pow_add, Nat.add_comm, Nat.mul_assoc, add_mul_div_left _ _ (two_pow_pos _), add_mod]
+  rw [Nat.pow_add, Nat.pow_one, Nat.mul_assoc]
+  simp
+
+theorem testBit_two_pow_mul_toNat_add {n w b} (h : n < 2 ^ w) :
+    testBit (2 ^ w * (bit b 0) + n) w = b := by
+  simp only [testBit, shiftRight_eq_div_pow]
+  rw [Nat.add_comm, add_mul_div_left _ _ (two_pow_pos _), Nat.div_eq_of_lt h, Nat.zero_add]
+  cases b <;> simp
+
+@[simp] theorem toNat_eq_bif {b: Bool} : cond b 1 0 = toNat b := by simp [cond, toNat]
+
+theorem shiftRight_eq_testBit (x i : Nat) : (x >>> i) % 2 = toNat (testBit x i) := by
+  simp [Nat.testBit, Nat.mod_two_of_bodd, toNat_eq_bif]
+
+theorem div_add_mod_two_pow (m n : Nat) : n = 2 ^ m * n >>> m + n % (2 ^ m):= by
```

Comment:
```suggestion
theorem div_add_mod_two_pow (m n : Nat) : n = 2 ^ m * n >>> m + n % (2 ^ m) := by
```
Should have shiftRight rather than div in its name!